dianna/methods/rise.py

```diff
@@ -115,7 +117,7 @@ def _get_predictions(self, sentences, runner, batch_size):
     def _create_masked_sentences(self, tokens, masks):
         tokens_masked = []
         for mask in masks:
-            tokens_masked.append(tokens[mask])
+            tokens_masked.append([token if keep else self.mask_string for token, keep in zip(tokens, mask)])
```

Comment:
I love this line!
I have to admit though, that it took a couple of minutes before it snapped into place in my mind. So on one hand it looks beautiful, but it isn't the clearest of lines. I could not think of a better alternative, other than extracting the list comprehension to its own function. This wouldn't solve much I think except for reducing the complexity on a single line. I am not sure if we should change it as it is.

Reply:
Thanks! I agree it's not the most readable but like you I couldn't think of a better way yet. I'll leave it as-is for now.